org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/CompletionTests.java

```diff
@@ -23395,7 +23402,11 @@ public void testBug346454c_2() throws JavaModelException {
 			"Test[METHOD_REF<CONSTRUCTOR>]{, Ltest.Test<>;, (I)V, Test, (i), " + (R_DEFAULT + R_RESOLVED + R_INTERESTING + R_NON_RESTRICTED) + "}\n" +
 			"Test[METHOD_REF<CONSTRUCTOR>]{, Ltest.Test<>;, (TT;)V, Test, (t), " + (R_DEFAULT + R_RESOLVED + R_INTERESTING + R_NON_RESTRICTED) + "}\n" +
 			"Test<>[ANONYMOUS_CLASS_DECLARATION]{, Ltest.Test<>;, (I)V, null, (i), " + (R_DEFAULT + R_RESOLVED + R_INTERESTING + R_NON_RESTRICTED) + "}\n" +
-			"Test<>[ANONYMOUS_CLASS_DECLARATION]{, Ltest.Test<>;, (TT;)V, null, (t), " + (R_DEFAULT + R_RESOLVED + R_INTERESTING + R_NON_RESTRICTED) + "}",
+			"Test<>[ANONYMOUS_CLASS_DECLARATION]{, Ltest.Test<>;, (TT;)V, null, (t), " + (R_DEFAULT + R_RESOLVED + R_INTERESTING + R_NON_RESTRICTED) + "}\n" +
+			"clone[METHOD_REF]{clone(), Ljava.lang.Object;, ()Ljava.lang.Object;, clone, null, " + (R_DEFAULT + R_RESOLVED + R_INTERESTING + R_CASE + R_UNQUALIFIED + R_NON_RESTRICTED) + "}\n" +
+			"getClass[METHOD_REF]{getClass(), Ljava.lang.Object;, ()Ljava.lang.Class;, getClass, null, " + (R_DEFAULT + R_RESOLVED + R_INTERESTING + R_CASE + R_UNQUALIFIED + R_NON_RESTRICTED) + "}\n" +
+			"hashCode[METHOD_REF]{hashCode(), Ljava.lang.Object;, ()I, hashCode, null, " + (R_DEFAULT + R_RESOLVED + R_INTERESTING + R_CASE + R_UNQUALIFIED + R_NON_RESTRICTED) + "}\n" + 
```

Comment:
@gayanper please double check your Eclipse settings: at this line the change introduced a trailing whitespace, which should never happen because the project has a save action configured which will remove all trailing white space.

I stumbled upon this when I edited the file and even after undo it contained a change wrt git index: the trailing white space was removed. I saw the same effect in another file (don't remember which and where), where it created several changes in different places, which makes creating a minimal patch for such file quite tedious. 

Thanks.

Reply:
> I saw the same effect in another file (don't remember which and where)

tripped over it again: `CompletionEngine` from the same commit.

Reply:
Sure i will check and fix the settings @stephan-herrmann in the workspace.

Reply:
Since this PR is merged, how do we handle it ?

Reply:
> Since this PR is merged, how do we handle it ?

I guess a dedicated commit only removing the trailing whitespace soon after opening for 4.29 would be best.

Did you find out how this could happen in the first place? The setting is actually persisted in git, so if `.settings/` is clean wrt git all should be good.

Invoking undo directly after saving will revert only the effect of save actions. Maybe that's what you did?

Reply:
I think it might be due to I import the projects as maven modules, But i will make sure the settings stays clean always to make sure i don't miss this.